grouper/user.py

```diff
@@ -57,3 +69,232 @@ def get_all_enabled_users(session):
         a list of all enabled User objects in the database
     """
     return session.query(User).filter_by(enabled=True).all()
+
+
+def enable_user(session, user, requester, preserve_membership):
+    """Enable a disabled user.
+
+    Args:
+        preserve_membership(bool): whether to remove user from any groups it may be a member of
+    Returns:
+        None
+    """
+    if not preserve_membership:
+        for group, group_edge in get_groups_by_user(session, user):
+            group_obj = session.query(Group).filter_by(
+                groupname=group.name
+            ).scalar()
+            if group_obj:
+                group_obj.revoke_member(
+                    requester, user, "group membership stripped as part of re-enabling account."
+                )
+
+    user.enabled = True
+    Counter.incr(session, "updates")
+
+
+def disable_user(session, user):
+    """Disables an enabled user"""
+    user.enabled = False
+    Counter.incr(session, "updates")
+
+
+def user_role_index(user, members):
+        if user.group_admin:
+            return GROUP_EDGE_ROLES.index("owner")
+        member = members.get(("User", user.name))
+        if not member:
+            return None
+        return member.role
+
+
+def user_role(user, members):
+    role_index = user_role_index(user, members)
+    if not role_index:
+        return None
+    else:
+        return GROUP_EDGE_ROLES[role_index]
+
+
+def user_has_permission(session, user, permission, argument=None):
```

Comment:
these new user + permission methods seem to make a logical group. thoughts on moving then into their own module `user_permissions` or some such?